docs/develop.md

```diff
@@ -0,0 +1,239 @@
+(on an Debian/APT based system):
+
+* build-essential
+* pkg-config
+* python-dev
+* python-virtualenv
+* libjpeg-dev
+* zlib1g-dev
+* libpng12-dev
+* libtiff5-dev
+* libfreetype6-dev
+* liblcms2-dev
+* libopenjpeg-dev
+* libwebp-dev
+* libpng12-dev
+* libxml2-dev
+* libxslt1-dev
+* liblzma-dev
+
+```shell
+apt-get install build-essential pkg-config python-dev python-virtualenv libjpeg-dev zlib1g-dev libpng12-dev libtiff5-dev libfreetype6-dev liblcms2-dev libopenjpeg-dev libwebp-dev libpng12-dev libxml2-dev libxslt1-dev liblzma-dev
+```
+
+## Python dependencies
+
+It is recommanded to work within a virtualenv to ensure proper dependencies isolation.
+
+It is recommanded to have a parent workspace where project are cloned, data stored...
+It will be designated by `$WORKSPACE` until the end of the documentation.
+
+```shell
+$ mkdir $WORKSPACE && cd $WORKSPACE
+```
+
+First, create and activate your virtualenv, with one of the following command
+(depending on you using virtualenv or virtualenvwrapper):
+
+```shell
+$ virtualenv . && source bin/activate  # for virtualenv
+$ mkvirtualenv udata && setvirtualenvproject .  # for virtualenvwrapper
+$ mkproject udata  # for virtualenvwrapper with configured project dir
+```
+
+If your system is using Python 3 as primary Python installation,
+don't forget to specify the python version with the `-p` parameter:
+
+```shell
+$ virtualenv -p /bin/python2 .
+$ mkvirtualenv udata -p /bin/python2
+$ mkproject -p /bin/python2 udata
+```
+
+Clone the uData project into your workspace:
+
+```shell
+$ git clone https://github.com/etalab/udata.git
+```
+
+Install the development dependencies and install the project as editable:
+
+```shell
+$ pip install -r udata/requirements/develop.pip
+$ pip install --no-deps -e udata/
+```
+
+## Middleware installation
+
+You can use native middleware packages or docker images (prefered method)
+
+### The docker way
+
+This is the prefered method as it does not depends of version provided by your OS.
+You need to have [Docker] installed and working for your user.
+
+You need to [install docker-compose]:
+
+```shell
+$ pip install docker-compose
+```
+
+A sample `docker-compose.yml` is provided in the udata repository.
+
+```shell
+$ cp udata/docker-compose.yml .
+$ docker-compose pull  # Pull docker images
+$ docker-compose up -d  # Run docker processes in background
+$ docjer-compose ps  # List running docker processes
+```
+
+ElasticSearch requires the elasticsearch-icu-analysis to be able to sort
+on unicode strings.
+You need to look at the compatibility matrix to find the corresponding version
+on [the official documention](https://github.com/elastic/elasticsearch-analysis-icu).
+At the time this doc is written, we use ElasticSearch 1.4.3 and ElasticSearch ICU Analysis 2.4.2
+
+```shell
+$ docker-compose run search /bin/bash
+$ /usr/share/elasticsearch/bin/plugin install elasticsearch/elasticsearch-analysis-icu/2.4.2
```

Comment:
We found a better way with @vinyll . I will change it.
We can simply run `docker-compose search run plugin install elasticsearch/elasticsearch-analysis-icu/2.4.2` so no need for a custom image.